### PR TITLE
Reorganize guide structure

### PR DESCRIPTION
**Pull request by bblaszkow06**

### CODE REVIEW

File: creating_element/tutorial.md [hunk 1]
```diff
@@ -41,24 +41,31 @@ def_input_pad :input,
   availability: :always,
   mode: :pull,
   demand_unit: :bytes,
-  caps: :any
+  caps: :any,
+  options: [
```
Comment (mat-hek) [inline creating_element/tutorial.md]: I think we shouldn't mention pad options if there are no dynamic pads here, but I leave the decision up to you

File: concepts/pads.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+Now, each time some element is linked to this pad, a new instance of the
+pad is created and callback [`handle_pad_added`](https://hexdocs.pm/membrane_core/Membrane.Element.Base.Mixin.CommonBehaviour.html#c:handle_pad_added/3)
+is invoked. Instances of a pad can be referenced as `{:dynamic, :input, number}`
+
+### Handling events
+
+What if Event such as End of Stream is passed through a pad of filter element?
+Usually if you are using pads `:input` and `:output` the default
+action is to forward an event. It means that if an event comes at `:input`
+pad it is set via `:output` pad and vice versa.
```
[the remark this answers is not in the document]
Comment (bblaszkow06) [reply concepts/pads.md]: Well, copy-paste might not have been the best idea #18 :wink:

File: concepts/pipelines.md [hunk 3]
```diff
@@ -1,5 +1,18 @@
 # Pipelines
 
-A pipeline is a container that consists of many elements and links between them. Like an Element, every Pipeline also has a playback state and on its basis, it manages the state of the contained elements.
+Conceptually, a pipeline is a container that consists of many elements and links between them.
+Like an element, every pipeline has a playback state (and uses the same callbacks to handle transitions between them).
+It's an Erlang process responsible for starting elements (usually referred to as children) and
```
Comment (mat-hek) [inline concepts/pipelines.md]: - Start with technically/physically would be a nice counterpart for this 'conceptually'
- End the sentence where now 'and' is, and put everything related to playback state in the next sentence

File: concepts/pipelines.md [hunk 4]
```diff
@@ -1,5 +1,18 @@
...
+managing their playback state to match the one set for the pipeline.
 
-During the application execution, elements may want to signal some events. For that purpose, they send the `Notification` to their supervisor, which in most cases is a pipeline. A programmer can handle those notifications by defining the appropriate method in the pipeline module.
+## Communication with children
+
+During the application execution, elements may want to signal some events. For that purpose, they send the
+`Notification` to their watcher, which in most cases is a pipeline.
+
+A pipeline can also a gateway to an element - as elements' pids aren't available globally pipeline can route messages
```
Comment (mat-hek) [inline concepts/pipelines.md]: can also _be?_ a gateway

**Pull request merged**